parser.y

```diff
@@ -7280,6 +7291,74 @@ SelectStmtFromTable:
 		$$ = st
 	}
 
+SelectStmtFromTableSample:
+	SelectStmtBasic "FROM" TableName PartitionNameListOpt TableAsNameOpt "TABLESAMPLE" TableSampleMethodOpt '(' Expression TableSampleUnitOpt ')' RepeatableOpt WhereClauseOptional
```

Comment:
it's better attach this to this line to support JOIN:

```bison
TableFactor:
	TableName PartitionNameListOpt TableAsNameOpt IndexHintListOpt TableSampleOpt # <--
```

this means the TableSample is part of the `ast.TableSource` or `ast.TableName`.

their use in UPDATE and DELETE can be rejected after parsing.